### PR TITLE
Added Null Checks in vocabulary.js

### LINKED ISSUE #199: [Bug] Missing Null Checks for Button and Panel Elements in Event Listeners
## Description
The current code that handles button clicks for expanding panels (exploreButton, explorePanel, menuButton, menuPanel) is missing null checks. Although there are null checks in place for the attributionButton and attributionPanel, the same checks are not implemented for other buttons and panels. This could lead to errors if these elements are not present in the DOM.

## Reproduction
1. Remove the exploreButton or menuButton from the HTML.
2. Run the script.
3. Observe the JavaScript error in the console related to null reference when trying to attach an event listener to a missing element.

## Expectation
The code should safely handle the case where any of the mentioned buttons or panels are not present in the DOM. By adding null checks similar to how it is done for attributionButton and attributionPanel, the JavaScript will be more robust and error-free in scenarios where elements might be missing.

## Environment

 - Device:  laptop
 - OS: Windows 11
 - Browser: Chrome
 - Version: 129.0.6668.100 
 - Other info: 1920 x 1080

## Resolution

- [X] I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by shivam8112005**
## Fixes

- Fixes #199 by @shivam8112005

## Description
The current code that handles button clicks for expanding panels (exploreButton, explorePanel, menuButton, menuPanel) is missing null checks. Although there are null checks in place for the attributionButton and attributionPanel, the same checks are not implemented for other buttons and panels. This could lead to errors if these elements are not present in the DOM.

## Technical details
This PR addresses the issue of missing null checks in the JavaScript code that manages various UI elements such as the explore panel, menu panel, and attribution panel. The updates ensure that event listeners are only attached when the corresponding elements are present in the DOM, improving the robustness of the code.

## Tests
1. Open the web application in a browser.
2. Open the developer console (F12).
3. Ensure there are no errors related to null or undefined when clicking on the explore, menu, or attribution buttons.

## Checklist

- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the *default* branch of the repository (`main` or `master`).
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by egbadon-victor** [comment]
Hi @shivam8112005   

Awesome effort here. However,  you’d submit a PR for an issue that has been marked `ready for work `.  This way your PRs can be considered. 

Here is a link to the contribution guidelines for reference [here](https://opensource.creativecommons.org/contributing-code/)

**Comment by shivam8112005** [comment]
@egbadon-victor thanks a lot for this info. now should i close this pr?

**Comment by possumbilities** [comment]
I'll second the note about submitted PRs too early above, but for this time the PR can remain open. Just keep this in mind on fuure Issues/PRs

**Comment by shivam8112005** [comment]
@possumbilities  ohk sure thank you.

File: src/js/vocabulary.js [hunk 1]
```diff
@@ -2,20 +2,25 @@ const exploreButton = document.querySelector('button.explore');
 const explorePanel = document.querySelector('.explore-panel');
 
 // explorePanel.classList.add('hide');
+if(exploreButton!==null && explorePanel!==null){
```
Comment (possumbilities) [inline src/js/vocabulary.js]: let's be sure there's a space between the parenthetical and opening bracket.

File: src/js/vocabulary.js [hunk 2]
```diff
@@ -2,20 +2,25 @@ const exploreButton = document.querySelector('button.explore');
...
 
 
 const menuButton = document.querySelector('button.expand-menu');
 const menuPanel = document.querySelector('.primary-menu');
 
-menuButton.addEventListener('click', (event) => {
-    menuPanel.classList.toggle('expand');
-    // explorePanel.classList.toggle('hide');
-});
+if(menuButton !== null && menuPanel !== null){
```
Comment (possumbilities) [inline src/js/vocabulary.js]: let's be sure there's a space between the parenthetical and opening bracket.

**Comment by possumbilities** [comment]
@shivam8112005 There's two minor spacing changes, but otherwise this is ready to move forward with Approval! Thankyou!

**Comment by shivam8112005** [comment]
@possumbilities  should i do one more commit to add space between them?

**Comment by shivam8112005** [comment]
You're welcome @possumbilities . I will add spacing between brackets once I reach home.

**Comment by shivam8112005** [comment]
done

**Review by possumbilities — APPROVED** [review]
Wonderful! Thank you. Approved.

**Pull request merged**